Add extra includes required for Boost 1.75 and later.

Taken from upstram:

  https://github.com/ceph/ceph/commit/ebf3a0398f18eab67d2ba25e6a10b41ff140f6a4

diff --git a/src/rgw/rgw_string.h b/src/rgw/rgw_string.h
index 257daa9c1fe6e..90e64f98a2587 100644
--- a/src/rgw/rgw_string.h
+++ b/src/rgw/rgw_string.h
@@ -8,5 +8,7 @@
 #include <stdlib.h>
 #include <limits.h>
 #include <string_view>
+#include <string>
+#include <stdexcept>
 
 #include <boost/container/small_vector.hpp>
